### ISSUE TITLE
Github OAuth integration

### ISSUE THREAD

**Issue by swilgosz** [driggl/linklog#18]
### Overview

Implement user logging in/registration using a Github API as an authorizer.

**Comment by swilgosz** [comment]
![image](https://user-images.githubusercontent.com/8088317/82885686-c252fd80-9f45-11ea-89d7-55d108a312ac.png)

@lusarz I'd added a Github OAuth application. When you'll start working on this ticket, send me a note to get credentials. 

I'm worried a bit that the API flow should be redesigned here a little bit.

Please let me know what endpoints exactly will you need to make this fully working.

Especially:

1. When user profile does not exist, register new one.
2. When user does exist, just log him in.

Currently in the API we have: [OAUTH login flow](https://documenter.getpostman.com/view/4406272/SVSGMq6b?version=latest#bf50ad58-ed5f-4ee6-aa01-6a9e5aa08ada)

Basically when you do the OAUTH, github should redirect you back with a one-time code to be used for authentication. From that point you can send this code to the API to register/login a user.

If there are any flaws with this flow just let me know and we'll adjust.
